awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -518,27 +587,21 @@ def create_if_needed(self, existing_item, new_item, endpoint, on_create=None, it
                     # User objects return username instead of name
                     self.json_output['name'] = response['json']['username']
                 self.json_output['id'] = response['json']['id']
+                self.json_output['created'] = True
                 self.json_output['changed'] = True
             else:
-                if 'json' in response and '__all__' in response['json']:
-                    self.fail_json(msg="Unable to create {0} {1}: {2}".format(item_type, item_name, response['json']['__all__'][0]))
-                elif 'json' in response:
-                    self.fail_json(msg="Unable to create {0} {1}: {2}".format(item_type, item_name, response['json']))
-                else:
-                    self.fail_json(msg="Unable to create {0} {1}: {2}".format(item_type, item_name, response['status_code']))
+                self.fail_json(msg="Unable to create {0} {1}: {2}".format(item_type, item_name, self.extract_errors_from_response(response)), **{'response': response})
 
-        # Process any associations with this item
-        if associations is not None:
-            for association_type in associations:
-                self.modify_associations(response, associations[association_type])
+        self.process_associations(response['json'], item_type, associations)
```

Comment:
This still isn't going to work if `existing_item` is non-null when the method is called.

https://github.com/ansible/awx/pull/6369

I'm sorry this is going to mess up the diff management even more. I really want us to focus on having these stable and testable as more features are added. If you want to put up a minor patch in a way that better aligns with with your direction, that's fine, but I want to address the minor patch first, before pulling large wide-ranging changes to the common code.